src/app/screens/Register/index.test.tsx

```diff
@@ -1,101 +1,63 @@
-  beforeAll(() => server.listen());
-  beforeEach(() => server.resetHandlers());
-  afterAll(() => server.close());
-
-  it('Render', async () => {
-    await render(<Register />);
-    expect(screen.getByRole('textbox', { name: AUTH_FIELDS.firstName })).toBeInTheDocument();
-    expect(screen.getByRole('textbox', { name: AUTH_FIELDS.lastName })).toBeInTheDocument();
-    expect(screen.getByRole('textbox', { name: AUTH_FIELDS.email })).toBeInTheDocument();
-    expect(screen.getByLabelText(AUTH_FIELDS.password)).toBeInTheDocument();
-    expect(screen.getByLabelText(AUTH_FIELDS.passwordConfirmation)).toBeInTheDocument();
-
-    const buttonRegister = screen.getByRole('button', { name: 'Register' });
-    expect(buttonRegister).toHaveAttribute('disabled');
+describe('Screen Register', () => {
+  it('Renders successfully', () => {
+    const { asFragment } = render(<Register />);
+    expect(asFragment()).toMatchSnapshot();
   });
 
-  it('Required fields', async () => {
-    await render(<Register />);
-    setInputByRole(AUTH_FIELDS.firstName, '');
-    setInputByRole(AUTH_FIELDS.lastName, '');
-    setInputByRole(AUTH_FIELDS.email, '');
-    setInputByLabelText(AUTH_FIELDS.password, '');
-    setInputByLabelText(AUTH_FIELDS.passwordConfirmation, '');
-
-    const buttonRegister = screen.getByRole('button', { name: 'Register' });
-    expect(buttonRegister).toHaveAttribute('disabled', '');
-
-    const alertErrorShow = 5;
-    expect(await screen.findAllByRole('alert')).toHaveLength(alertErrorShow);
-  });
+  it('If the form is not completed correctly, it is not sent', async () => {
+    const mockSave = jest.fn();
 
-  it('Field validation', async () => {
-    await render(<Register />);
-    setInputByRole(AUTH_FIELDS.email, 'name');
-    setInputByLabelText(AUTH_FIELDS.password, 'asdf');
-    setInputByLabelText(AUTH_FIELDS.passwordConfirmation, 'fdsa');
+    render(<Register mockFunction={mockSave} />);
 
-    const buttonRegister = screen.getByRole('button', { name: 'Register' });
-    expect(buttonRegister).toHaveAttribute('disabled', '');
+    const buttonRegister = screen.getByRole('button', { name: I18N_KEYS.buttonRegister });
+    fireEvent.click(buttonRegister);
 
-    const alertErrorShow = 2;
+    const alertErrorShow = 5;
     expect(await screen.findAllByRole('alert')).toHaveLength(alertErrorShow);
-  });
-
-  it('Form filled out without errors', async () => {
-    await render(<Register />);
-    setInputByRole(AUTH_FIELDS.firstName, USER_TEST.firstName);
-    setInputByRole(AUTH_FIELDS.lastName, USER_TEST.lastName);
-    setInputByRole(AUTH_FIELDS.email, USER_TEST.email);
-    setInputByLabelText(AUTH_FIELDS.password, USER_TEST.password);
-    setInputByLabelText(AUTH_FIELDS.passwordConfirmation, USER_TEST.passwordConfirmation);
-
-    const buttonRegister = screen.getByRole('button', { name: 'Register' });
-    expect(buttonRegister).not.toHaveAttribute('disabled', false);
+    expect(mockSave).not.toBeCalled();
```

Comment:
what you could do is split your register screen, the register screen could have the onSubmit function, and create a component `RegisterForm` that contains the form and receive the onSubmit as prop.
In that way you could mock the onSubmit and send that as prop to check if is not called